sui_types/src/message_headers.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
+pub enum TraceTag {
+    LatencyProbe,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Header {
+    TraceId(u64),
+    TraceTag(TraceTag),
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct SerializedHeaders {
+    // Zero or more Header instances. Only the first
+    // occurrence of a variant is read, subsequent ones
+    // are discarded.
+    pub headers: Vec<Header>,
```

Comment:
ok, tell me that the common case if for this to be empty, which leads to no (additional) allocation on each message, right?

Reply:
Correct!